Feat: initial implementation for single artifact release

hfend opened the pull request:
This PR introduces an initial working implementation for a Keygen.sh Release action.

hfend committed:
Initial implementation for single artifact release

hfend committed:
Update README AND CONTRIBUTING

hfend committed:
Update note on file upload memory considerations

hfend committed:
Update readme (add todos)

hfend committed:
Update readme

hfend committed:
Support multiple artifacts

hfend committed:
Update readme todos

morgante commented on __tests__/main.test.ts:
```diff
@@ -1,29 +1,6 @@
 import {expect, test} from '@jest/globals'
 
-test('throws invalid number', async () => {
+test('dummy', async () => {
```
Tests would be nice!

hfend replied:
Added a happy path test in https://github.com/unhack/keygen-action/pull/11/commits/ff9992ae9409113f3d7c8a0d5b90e98ad606184b

morgante commented on src/api/index.ts:
```diff
@@ -0,0 +1,184 @@
+    // As a result, the upload operation is at the mercy of available Runner memory.
+    //
+    // We would've made the upload read the file as a stream (using fs.createReadStream()). However, S3 does not support this.
+    // We would've also considered supporting S3's Multipart upload (when a file is detected to be larger than a given threshold). Howeber, Keygen does not provide utilities for getting pre-signed part upload urls.
```
What does S3 not support about it? Can't you send a streaming HTTP PUT regardless of what the API supports - this seems like a major limitation?

hfend replied:
My bad, S3 supports streaming uploads just fine. Implemented in https://github.com/unhack/keygen-action/pull/11/commits/00251801f5a0eeb019bb7f1effbeac09d297e943

morgante reviewed (commented):
Nice job! 🥳

hfend committed:
Support streaming uploads

morgante commented:
Looks like checks failed.

hfend committed:
Test happy path

hfend committed:
Fix check-dist.yml workflow

hfend committed:
Fix test.yml workflow
Made it rely on the jest test
Removed the Action test as that would require setting up a dummy Keygen environment

hfend committed:
Lint main.test.ts

hfend commented:
> Looks like checks failed.

Made fixes in https://github.com/unhack/keygen-action/pull/11/commits/61ff24f31a5ccdae0573bf8089104eedefa55b3b and https://github.com/unhack/keygen-action/pull/11/commits/460234213457f13354030944c6f570a0e3b4431b
